src/components/HdModal.vue

```diff
@@ -1,128 +1,93 @@
-            :class="{
-              'btn--primary': action.type === 'primary',
-              'btn--ghost': action.type !== 'primary',
-            }"
-            class="modal__actions__action btn"
-            type="button"
-            @click="action.callback"
+          <div
+            class="hd-modal__action"
+            v-for="(button, name) in actions"
+            :key="`action-${name}`"
           >
-            {{ action.label }}
-          </button>
-        </div>
+            <slot :name="name" :value="button">
+              <hd-button
+                :modifier="button.modifier"
+                :is-in-dark-background="button.isInDarkBackground"
+                :disabled="button.disabled"
+                :icon-src="button.iconSrc"
+                @click="$emit('action', button.name)"
+              >
+                {{ button.text }}
+              </hd-button>
+            </slot>
+          </div>
+        </footer>
+
         <button
           v-if="showCloseIcon"
-          :class="`modal__close--${closeIconColor}`"
-          class="modal__close"
-          @click="onCloseClick"
+          class="hd-modal__close-button"
+          @click="$emit('close')"
         >
-          <HdIcon
+          <hd-icon
             :src="closeIcon"
-            width="100%"
+            class="hd-modal__close-icon"
           />
         </button>
       </div>
-    </section>
+    </article>
   </transition>
 </template>
 
 <script>
+import HdButton from 'homeday-blocks/src/components/buttons/HdButton.vue';
 import HdIcon from 'homeday-blocks/src/components/HdIcon.vue';
 import { close as closeIcon } from 'homeday-assets';
+import { getMessages } from 'homeday-blocks/src/lang';
 
 export default {
   name: 'HdModal',
   components: {
+    HdButton,
     HdIcon,
   },
   props: {
-    closeIconColor: {
-      type: String,
-      default: 'dark',
-      validator: (closeIconColor) => ['dark', 'light'].includes(closeIconColor),
-    },
-    showCloseIcon: {
-      type: Boolean,
-      default: true,
-    },
-    externalCloseIcon: {
-      type: Boolean,
-      default: false,
-    },
-    overlayColor: {
-      type: String,
-      default: 'dark',
-      validator: (overlayColor) => ['dark', 'light'].includes(overlayColor),
-    },
-    width: {
-      type: Number,
-      default: 0,
-      validator: (maxWidth) => maxWidth >= 0,
-    },
-    spacingSize: {
+    iconSrc: {
       type: String,
-      default: 'large',
-      validator: (spacingSize) => ['large', 'normal'].includes(spacingSize),
+      default: '',
     },
-    noSpacing: {
+    isWide: {
       type: Boolean,
       default: false,
     },
-    noExternalSpacing: {
+    showCloseIcon: {
       type: Boolean,
-      default: false,
+      default: true,
     },
-    mobileAlign: {
+    lang: {
       type: String,
-      default: 'center',
-      validator: (mobileAlign) => ['center', 'bottom'].includes(mobileAlign),
+      default: 'de',
     },
     actions: {
```

Comment:
For the new solution to work properly, the all the entries of `actions` should have a "name" attribute.
To make this clear for the developer, I suggest to add a custom [prop validator](https://vuejs.org/v2/guide/components-props.html#Prop-Validation) here. Something like: `(entry) => typeof entry.name == 'string';` can do the job, you can also show a warning to explain why it's not valid (e.g. `The attribute "name" is required`)

Reply:
https://github.com/homeday-de/homeday-blocks/pull/732/commits/397b212782decb1480f40b224cab14fc9303f453

Reply:
Very nice @volcanioo,
Just one little improvement to your implementation:
You can make use of the `validator` function even more, by returning `false` if there is an issue with the prop, printing the error is good, but it's even better to use what Vue provides (so it can show the error to the developer in a better way and also fail the build if needed). You can write the validator like this:

```js
validator: (actions) => {
  const indexOfActionMissingName = actions.findIndex((action) => typeof action.name !== 'string');
  
  if (indexOfActionMissingName >= 0) {
    console.warn(`The attribute "name" is required on actions[${indexOfActionMissingName}]`);
    return false;
  }
  
  return true;
},
```

Reply:
I understand what you meant thank you again man  https://github.com/homeday-de/homeday-blocks/pull/732/commits/dd0e5434436bc0ea4512a89baccfaf833bd3e061

Reply:
🚀